### PR TITLE
fix: show user friendly error message when user publish past event

### LINKED ISSUE #1854: Show user friendly error message when user tries to publish a past event
**Describe the bug**
Show user friendly error message when user tries to publish a past event.

**Would you like to work on the issue?**
Open to all.

### PR DESCRIPTION
**Pull request by 991rajat**
Fixes #1854 

Checklist:

- [x] I have checked for PMD and check-style issues 
- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream development branch.

Changes: [Add here what changes were made in this issue and if possible provide links.]

Screenshots for the change:

### CODE REVIEW

**Comment by iamareebjamal** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-app/pullRequest?prid=4963535&bid=16403932)

File: app/src/main/java/com/eventyay/organizer/core/event/dashboard/EventDashboardPresenter.java [hunk 1]
```diff
@@ -134,7 +135,13 @@ public void toggleState() {
         event.state = Event.STATE_DRAFT.equals(event.state) ? Event.STATE_PUBLISHED : Event.STATE_DRAFT;
         eventRepository.updateEvent(event)
             .compose(dispose(getDisposable()))
-            .compose(progressiveErroneous(getView()))
+            .compose(ViewTransformers.progressive(getView()))
+            .doOnError(throwable -> {
+                if(ErrorUtils.getErrorDetails(throwable).getPointer().equals("starts-at"))
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/event/dashboard/EventDashboardPresenter.java]: Codacy Issue found: ['if' is not followed by whitespace.](https://app.codacy.com/manual/fossasia/open-event-orga-app/pullRequest?prid=4963535)

Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/event/dashboard/EventDashboardPresenter.java]: Codacy Issue found: ['if' construct must use '{}'s.](https://app.codacy.com/manual/fossasia/open-event-orga-app/pullRequest?prid=4963535)

**Comment by 991rajat** [comment]
@iamareebjamal need help with build failed.

**Comment by iamareebjamal** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-app/pullRequest?prid=4963535&bid=16403932)

File: app/src/main/java/com/eventyay/organizer/core/event/dashboard/EventDashboardFragment.java [hunk 2]
```diff
@@ -208,7 +208,11 @@ public void showOrderStatistics(OrderStatistics orderStatistics) {
 
     @Override
     public void showError(String error) {
-        ViewUtils.showSnackbar(binding.getRoot(), error);
+        if (error.contains("starts-at")){
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/event/dashboard/EventDashboardFragment.java]: Codacy Issue found: ['{' is not preceded with whitespace.](https://app.codacy.com/manual/fossasia/open-event-orga-app/pullRequest?prid=4963535)

**Comment by iamareebjamal** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-app/pullRequest?prid=4963535&bid=16403932)

File: app/src/main/java/com/eventyay/organizer/core/event/dashboard/EventDashboardFragment.java [hunk 3]
```diff
@@ -208,7 +208,11 @@ public void showOrderStatistics(OrderStatistics orderStatistics) {
 
     @Override
     public void showError(String error) {
-        ViewUtils.showSnackbar(binding.getRoot(), error);
+        if (error.equals("starts-at should be after current date-time - starts-at")) {
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/event/dashboard/EventDashboardFragment.java]: Codacy Issue found: [Position literals first in String comparisons](https://app.codacy.com/manual/fossasia/open-event-orga-app/pullRequest?prid=4963535)

File: app/src/main/java/com/eventyay/organizer/core/event/dashboard/EventDashboardFragment.java [hunk 4]
```diff
@@ -208,7 +208,11 @@ public void showOrderStatistics(OrderStatistics orderStatistics) {
 
     @Override
     public void showError(String error) {
-        ViewUtils.showSnackbar(binding.getRoot(), error);
+        if (error.contains("starts-at")) {
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/event/dashboard/EventDashboardFragment.java]: You need to properly parse the error

Comment (991rajat) [reply app/src/main/java/com/eventyay/organizer/core/event/dashboard/EventDashboardFragment.java]: Not able to understand. Pls, elaborate.

Comment (iamareebjamal) [reply app/src/main/java/com/eventyay/organizer/core/event/dashboard/EventDashboardFragment.java]: starts-at can be in any error unrelated to past events

Comment (991rajat) [reply app/src/main/java/com/eventyay/organizer/core/event/dashboard/EventDashboardFragment.java]: Do I need to match the exact error coming from the backend?

Comment (991rajat) [reply app/src/main/java/com/eventyay/organizer/core/event/dashboard/EventDashboardFragment.java]: OK

**Comment by 991rajat** [comment]
Is message is ok?
> Past event cannot be updated.
Or should I change to 
> Past event cannot be published.

**Comment by iamareebjamal** [comment]
published

**Comment by 991rajat** [comment]
@iamareebjamal Pls review the code.

**Comment by iamareebjamal** [comment]
It's fine. Use strings.xml

**Comment by 991rajat** [comment]
APK
[app.zip](https://github.com/fossasia/open-event-organizer-android/files/4174516/app.zip)

File: app/src/main/java/com/eventyay/organizer/core/event/dashboard/EventDashboardFragment.java [hunk 5]
```diff
@@ -208,7 +208,11 @@ public void showOrderStatistics(OrderStatistics orderStatistics) {
 
     @Override
     public void showError(String error) {
-        ViewUtils.showSnackbar(binding.getRoot(), error);
+        if (error.contains(getString(R.string.past_event_error_message))) {
+            ViewUtils.showSnackbar(binding.getRoot(), "Past events cannot be published.");
```
Comment (991rajat) [inline app/src/main/java/com/eventyay/organizer/core/event/dashboard/EventDashboardFragment.java]: @iamareebjamal "Past events cannot be published." Is this should be put inside strings too and do I need to also convert it into the Chinese language too?

Comment (iamareebjamal) [reply app/src/main/java/com/eventyay/organizer/core/event/dashboard/EventDashboardFragment.java]: You NEVER, EVER should put something you match programmatically in strings.xml

Remove it and put this in strings.xml

Comment (991rajat) [reply app/src/main/java/com/eventyay/organizer/core/event/dashboard/EventDashboardFragment.java]: OK, Thanks

**Comment by 991rajat** [comment]
@iamareebjamal Pls review, Thanks

**Review by iamareebjamal — APPROVED** [review]

**Pull request merged**